pkg/apis/core/types.go

```diff
@@ -3948,6 +3951,16 @@ type LocalObjectReference struct {
 	Name string
 }
 
+// TypedLocalObjectReference contains enough information to let you locate the typed referenced object inside the same namespace.
+type TypedLocalObjectReference struct {
+	// Name of the referent.
+	// +optional
+	Name string
+	// Kind of the referent.
+	// +optional
+	Kind string
```

Comment:
@lavalamp @kubernetes/sig-api-machinery-api-reviews  I need a consult.

Is "kind" sufficient here?  In general a kind holds something like "Endpoints" while apiVersion holds "extensions/v1beta1".  In a context like this, I don't care which API version, but I do care which group.

Reply:
kind, API group, and name are the three bits of info needed to unambiguously reference a particular object (assuming you don't care what version of the object you deal with).

for example:
https://github.com/kubernetes/kubernetes/blob/7749eeaf3f1eddd0796a4d61e545a90e7d9d59b0/staging/src/k8s.io/api/rbac/v1/types.go#L90-L98

Reply:
Thanks, Jordan, that's what I assumed.  I saw that ObjectReference has a field "APIVersion" which seemed wrong...  I didn't know this struct, but it makes more sense.

Reply:
Added APIGroup.

Reply:
Then we will add the APIGroup in the struct. I assumed it is required and user has to specify the APIGroup here for the data source.